tests/foreman/ui/test_host.py

```diff
@@ -2311,28 +2311,57 @@ def test_positive_gce_cloudinit_provision_end_to_end(
             googleclient.disconnect()
 
 
-@pytest.mark.destructive
 @pytest.mark.upgrade
-@pytest.mark.usefixtures('install_cockpit_plugin')
 @pytest.mark.tier2
-def test_positive_cockpit(session, default_sat):
+def test_positive_cockpit(destructive_sat):
     """Install cockpit plugin and test whether webconsole button and cockpit integration works
 
     :id: 5a9be063-cdc4-43ce-91b9-7608fbebf8bb
 
     :expectedresults: Cockpit page is loaded and displays sat host info
 
+    :BZ: 1876220
+
     :CaseLevel: System
 
+    :steps:
+        1. register the destructive_sat to dogfood and install cockpit on it.
+        2. kill the cockpit service.
+        3. go to web console and verify if getting 503 error.
+        4. "satellite-maintain service list" to check if service "cockpit.service" exists.
+        5. "satellite-maintain service restart" to restart the satellite services.
+        6. check Cockpit page is loaded and displays sat host info.
     """
-    with session:
-        session.organization.select(org_name='Default Organization')
-        session.location.select(loc_name='Any Location')
-        hostname_inside_cockpit = session.host.get_webconsole_content(
-            entity_name=default_sat.hostname
-        )
-        assert (
-            hostname_inside_cockpit == default_sat.hostname
-        ), 'cockpit page shows hostname {} instead of {}'.format(
-            hostname_inside_cockpit, default_sat.hostname
-        )
+
+    destructive_sat.register_to_dogfood()
+    destructive_sat.install_cockpit()
```

Comment:
Since this fixture is only used by this test at the moment so I thought of placing directly in this test rather than in fixture.
But yeah, I'll move it back to fixture.